.circleci/config.yml

```diff
@@ -47,28 +54,56 @@ jobs:
             name: Build Docker images
             working_directory: /ichnaea
             command: |
-                # consolidated from a bunch of commands in the older format
-                I="image-$(date +%j).gz"
-                if [[ -e ~/docker/$I ]]
-                    then echo "Loading $I"
-                    pigz -d -c ~/docker/$I | docker load
-                fi
-                docker build -t ${DOCKERHUB_REPO:-mozilla/location} .
-                mkdir -p ~/docker
-                rm -f ~/docker/*
-                docker save ${DOCKERHUB_REPO:-mozilla/location} | pigz --fast -c > ~/docker/$I
-                ls -l ~/docker
-
-        # write the sha256 sum to an artifact to make image verification easier
-        - run: docker images --no-trunc | awk '/^app/ {print $3}' | tee /ichnaea/docker-image-shasum256.txt
-        - store_artifacts:
-            path: /ichnaea/docker-image-shasum256.txt
+                docker build -t mozilla/location .
 
         - run:
             name: Run tests
-            path: /ichnaea
+            working_directory: /ichnaea
             command: |
                 ./dev test
 
-        - store_test_results:
-            path: /tmp/circleci-test-results
+        - run:
+            name: Push to Dockerhub
+            working_directory: /ichnaea
+            command: |
+              # Quit early if docker credentials are not defined
+              if [ "${DOCKER_USERNAME}" == "" ] || [ "${DOCKER_PASSWORD}" == "" ]; then
+                echo "Skipping Push to Dockerhub, credentials not available."
+                exit 0
+              fi
+
+              function retry {
+                set +e
+                local n=0
+                local max=3
+                while true; do
+                "$@" && break || {
+                  if [[ $n -lt $max ]]; then
+                    ((n++))
+                    echo "Command failed. Attempt $n/$max:"
+                  else
+                    echo "Failed after $n attempts."
+                    exit 1
+                  fi
+                }
+                done
+                set -e
+              }
+
+              export DOCKER_TAG="${CIRCLE_SHA1}"
+              if [ -n "${CIRCLE_TAG}" ]; then
+                export DOCKER_TAG="${CIRCLE_TAG}"
+              fi
+              # push on master or git tag
+              if [ "${CIRCLE_BRANCH}" == "master" ] || [ -n "${CIRCLE_TAG}" ]; then
+                echo "${DOCKER_PASSWORD}" | docker login -u="${DOCKER_USERNAME}" --password-stdin
+                docker tag "mozilla/location" "mozilla/location:${DOCKER_TAG}"
```

Comment:
In the Socorro version, this is also retried. I have vague memories of this failing at least once.

Reply:
On second thought, looks like I made it retry in Socorro and I don't know offhand why I would have done that, so maybe let's ignore what I just said.

Reply:
> In the Socorro version, this is also retried. I have vague memories of this failing at least once. Having said that, I'm not sure what circumstances would cause this to fail and retrying would be good.

Reply:
- - [ ] # []()~~@@[]()__****__ ~~

Reply:
> > In the Socorro version, this is also retried. I have vague memories of this failing at least once. Having said that, I'm not sure what circumstances would cause this to fail and retrying would be good.
> y
>